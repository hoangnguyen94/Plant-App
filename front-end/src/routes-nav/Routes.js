//Route Component
import * as React from "react";
import { Switch, Route, Redirect } from "react-router-dom";
import Homepage from "../homepage/Homepage";
import LoginForm from "../auth/LoginForm";
import ProfileForm from "../profiles/ProfileForm";
import SignupForm from "../auth/SignupForm";
import PrivateRoute from "./PrivateRoute";
import PlantCard from "../plant/PlantCard";


/** Site-wide routes.
 *
 * Parts of site should only be visitable when logged in. Those routes are
 * wrapped by <PrivateRoute>, which is an authorization component.
 *
 * Visiting a non-existant route redirects to the homepage.
 */

function Routes({ login, signup, selectedPlantData }) {
  console.debug(
      "Routes",
      `login=${typeof login}`,
      `signup=${typeof signup}`,
  );

  return (
    <div className="pt-5">
      <Switch>

        <Route exact path="/">
          <Homepage />
        </Route>

        <Route exact path="/login">
          <LoginForm login={login} />
        </Route>

        <Route exact path="/signup">
          <SignupForm signup={signup} />
        </Route>

        <PrivateRoute exact path="/plants"> 
          <PlantCard selectedPlantData={selectedPlantData}/>
        </PrivateRoute>
        
        <PrivateRoute path="/profile">
          <ProfileForm />
        </PrivateRoute>

        <Redirect to="/" />

      </Switch>

      
      </div>
  );
}

export default Routes;
