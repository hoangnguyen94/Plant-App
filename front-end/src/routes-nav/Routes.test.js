import React from "react";
import { render } from "@testing-library/react";
import Routes from "./Routes";
import { MemoryRouter } from "react-router";
import UserContext from "../auth/UserContext";

const mockUser = {
	email: "testing@test.com",
	firstName: "test",
	lastName: "test",
	isAdmin: false,
	username: "testing"
};
  
it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <UserContext.Provider value={{ currentUser: mockUser }}>
        <Routes />
      </UserContext.Provider>
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserContext.Provider value={{ currentUser: mockUser }}>
        <Routes />
      </UserContext.Provider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
