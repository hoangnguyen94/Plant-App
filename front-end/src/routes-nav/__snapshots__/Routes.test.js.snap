// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="pt-5"
  >
    <div
      class="Homepage"
    >
      <div
        class="container text-center"
      >
        <h1
          class="mb-4 font-weight-bold"
        >
          Plant
        </h1>
        <p
          class="lead"
        >
          When you plant something, you invest in a beautiful future amidst a stressful, chaotic and, at times, downright appalling world.
        </p>
        <h2>
          Welcome Back, test!
        </h2>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
